### PR TITLE
Optimizes the `output_part_info()` proc in Mech Fabricators.

### PR DESCRIPTION
**Pull request by Zonespace27**
## About The Pull Request
It turns out the mech fabricator updated the list of what was available every time anyone opened it, which is incredibly expensive.

In a round, it was the 9th most CPU-overtime intensive proc: (/obj/machinery/mecha_part_fabricator/proc/output_part_info())
image

I optimized it to only update the list on Init and when the console is re-synced with the R&D console, in addition to making the proc much, much lighter to run.

Local Testing:
Called the old version 1000 times.
image

Called the new version 2000 times.
image

(Yes I tested this, but it was on a downstream, so I did the changes through Github)
## Why It's Good For The Game

Mecha fabricators' menu update proc shouldn't be the 9th most CPU-overloaded proc in an active round, never mind being called 768 times in less than an hour.

## Changelog

:cl:
refactor: Mech Fabricator's `output_part_info()` proc is much more optimized.
/:cl:

### CODE REVIEW

**Comment by Mothblocks** [comment]
The real solution is to make this a `/datum/asset/json` like prefs menu and circuits menu is, so that it only ever happens truly *once*, and is just made an asset.

But that requires some more code on the client end, and I still need to generalize that.

File: code/modules/vehicles/mecha/mech_fabricator.dm [hunk 1]
```diff
@@ -142,11 +149,8 @@
 					sub_category += "Engineering"
 			else
 				sub_category += "All Cyborgs"
-		else if(ispath(built_item, /obj/item/borg_restart_board))
-			sub_category += "All Cyborgs" //Otherwise the restart board shows in the "parts" category, which seems dumb
```
Comment (LemonInTheDark) [inline code/modules/vehicles/mecha/mech_fabricator.dm:146]: Where did this go?

**Comment by LemonInTheDark** [comment]
CPU intensive isn't quite the right term for it, overtime is bad in a different sort of way

File: code/modules/vehicles/mecha/mech_fabricator.dm (hunk 1 above)
Comment (Zonespace27) [reply code/modules/vehicles/mecha/mech_fabricator.dm:146]: Readded it, not sure how I missed it originally

File: code/modules/vehicles/mecha/mech_fabricator.dm [hunk 2]
```diff
@@ -181,6 +188,35 @@
...
 	return part
 
+/**
+ * Updates the `final_sets` and `buildable_parts` for the current mecha fabricator.
+ */
+/obj/machinery/mecha_part_fabricator/proc/update_menu_tech()
+	final_sets = list()
+	buildable_parts = list()
+	for(var/part_set in part_sets)
+		final_sets += part_set
```
Comment (LemonInTheDark) [inline code/modules/vehicles/mecha/mech_fabricator.dm:198]: Why not just do 
```suggestion
	final_sets += part_sets
```

File: code/modules/vehicles/mecha/mech_fabricator.dm [hunk 3]
```diff
@@ -194,8 +194,7 @@
 /obj/machinery/mecha_part_fabricator/proc/update_menu_tech()
 	final_sets = list()
 	buildable_parts = list()
-	for(var/part_set in part_sets)
-		final_sets += part_set
+	final_sets += part_sets
```
Comment (Mothblocks) [inline code/modules/vehicles/mecha/mech_fabricator.dm:197]: you couldve `final_sets = part_sets.Copy()` but whatever

**Pull request merged**